### PR TITLE
Cleanup Akka.Cluster.SBR for release

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Building on the work done in https://github.com/akkadotnet/akka.net/pull/4587 and cleaning up code, updating documenation, working on a clean cut-over path from the current Akka.NET split brain resolver. Delving into the code more some too.

### CODE REVIEW

File: docs/articles/clustering/split-brain-resolver.md [hunk 1]
```diff
@@ -37,8 +34,81 @@ akka.cluster {
...
+* `lease-majority`
+* `keep-referee` - only available with the legacy split brain resolver.
+
+All strategies will be applied only after cluster state has reached stability for specified time threshold (no nodes transitioning between different states for some time), specified by `stable-after` setting. Nodes which are joining will not affect this treshold, as they won't be promoted to UP status in face unreachable nodes. For the same reason they won't be taken into account, when a strategy will be applied.
 
-## Strategies
+```hocon
+akka.cluster.split-brain-resolver {
+  # Enable one of the available strategies (see descriptions below):
+  # static-quorum, keep-majority, keep-oldest, keep-referee
```
Comment (IgorFedchenko) [inline docs/articles/clustering/split-brain-resolver.md:71]: There are `down-all` and `lease-majority` strategies listed above - do we need to also list them here?

**Review by IgorFedchenko — APPROVED** [review]
Left one small notice - but in general, LGTM

File: docs/articles/clustering/split-brain-resolver.md (hunk 1 above)
Comment (Aaronontheweb) [reply docs/articles/clustering/split-brain-resolver.md:71]: Yes, you're right - I should add some material to describe those. I'll update this PR accordingly.

Comment (Aaronontheweb) [reply docs/articles/clustering/split-brain-resolver.md:72]: Done

**Pull request merged**